src/components/questionnaire/questionnaire-manager.tsx

```diff
@@ -0,0 +1,262 @@
+  const { id } = useParams<{ id: string }>();
+  const { questionnairesApi } = useLambdasApi();
+  const setError = useSetAtom(errorAtom);
+  const [questionnaire, setQuestionnaire] = useState<Questionnaire>({
+    id: "",
+    title: "",
+    description: "",
+    questions: [],
+    passScore: 0
+  });
+  const [loading, setLoading] = useState(true);
+  const [userResponses, setUserResponses] = useState<UserResponses>({});
+  const [ifPassedMessage, setIfPassedMessage] = useState<string | null>(null);
+  const [passedDialogOpen, setPassedDialogOpen] = useState<boolean>(false);
+  const users = useAtomValue(usersAtom);
+  const userProfile = useAtomValue(userProfileAtom);
+  const navigate = useNavigate();
+
+  const loggedInUser = users.find((user: User) => user.id === userProfile?.id);
+
+  useEffect(() => {
+    const fetchQuestionnaire = async () => {
+      if (!id) return;
+      setLoading(true);
+      try {
+        const fetchedQuestionnaire = await questionnairesApi.getQuestionnairesById({ id });
+        setQuestionnaire(fetchedQuestionnaire);
+      } catch (error) {
+        setError(`${strings.error.questionnaireLoadFailed}, ${error}`);
+      }
+      setLoading(false);
+    };
+    fetchQuestionnaire();
+  }, [id, questionnairesApi, setError]);
+
+  if (loading) {
+    return (
+      <CircularProgress
+        size={50}
+        style={{
+          position: "absolute",
+          top: "50%",
+          left: "50%",
+          transform: "translate(-50%, -50%)"
+        }}
+      />
+    );
+  }
+
+  /**
+   * Function to handle the change of the answer option checkboxes
+   *
+   * @param questionText
+   * @param answerLabel
+   * @param isSelected
+   */
+  const handleCheckboxChange = (questionText: string, answerLabel: string, isSelected: boolean) => {
+    setUserResponses((prevResponses) => {
+      const selectedAnswerLabels = prevResponses[questionText] || [];
+      return {
+        ...prevResponses,
+        [questionText]: isSelected
+          ? [...selectedAnswerLabels, answerLabel]
+          : selectedAnswerLabels.filter((label) => label !== answerLabel)
+      };
+    });
+  };
+
+  /**
+   * Function to handle the change of the answer option radio buttons
+   *
+   * @param questionText
+   * @param answerLabel
+   */
+  const handleRadioChange = (questionText: string, answerLabel: string) => {
+    setUserResponses((prevResponses) => ({
+      ...prevResponses,
+      [questionText]: [answerLabel]
+    }));
+  };
+
+  /**
+   * Function to handle the submission of the questionnaire
+   */
+  const handleSubmit = async () => {
+    let correctAnswersCount = 0;
+    questionnaire.questions?.forEach((question) => {
+      const userAnswers = userResponses[question.questionText] || [];
+      const correctAnswers = question.answerOptions
+        .filter((option) => option.isCorrect)
+        .map((option) => option.label);
+      if (
+        userAnswers.length === correctAnswers.length &&
```

Comment:
Actually that is true. As we cannot give "extra" answers users lenght cannot be over correctAnswers lenght. 

At now that is the case. We discuss this at begining when this boolean solution was selected. At this point it might have reguired bit more thought.
Same thing with question to have own Id.